src/components/Swap/SwapActionButton/index.tsx

```diff
@@ -28,8 +28,8 @@ export default function SwapActionButton() {
       (!permit2Enabled && approval.state !== SwapApprovalState.APPROVED) ||
       error !== undefined ||
       (!isWrap && !trade) ||
-      !(inputCurrencyAmount && inputCurrencyBalance) ||
-      inputCurrencyBalance.lessThan(inputCurrencyAmount),
+      !inputCurrencyAmount ||
```

Comment:
i'm not sure if we still need this check - if we're requiring `inputCurrencyAmount`, shouldn't we also be requiring `outputCurrencyAmount`?

Reply:
We're not checking it against a balance, so we don't need to check it at all: `!(inputCurrencyAmount && inputCurrencyBalance)` acted as a guard to ensure that `inputCurrencyBalance.lessThan(inputCurrencyAmount)` was valid. I _think_ this should be covered by `(!isWrap && !trade)`.

Reply:
this is true, but we avoid some type warnings if we keep it.